api/include/opentelemetry/trace/semantic_conventions.h

```diff
@@ -0,0 +1,190 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+#define OTEL_TRACE_GENERATE_SEMCONV_METHOD(attribute_name, attribute_value) \
+  static const nostd::string_view Get##attribute_name() noexcept            \
```

Comment:
Ideally this should be expressible as `constexpr` - to return the value of it at compile time, without incurring the overhead of a static method call. I'm not sure if the getters give significant benefit, as in - better structure or something? From Google C++ style document: `Global strings: if you require a global or static string constant, consider using a simple character array, or a char pointer to the first element of a string literal. String literals have static storage duration already and are usually sufficient.`

Reply:
How can we do it in an all header way? C++11 won't allow initializing`std::string`/`nostd::string` within class definition.

Reply:
You just don't use getters. Use constants with `constexpr` definition?

Reply:
Or if you use getters, return straight the literal constant type without any class:

![image](https://user-images.githubusercontent.com/34072974/122806003-a6bc4280-d27e-11eb-96da-6a0d9a0dc675.png)

Reply:
> Or if you use getters, return straight the literal constant type without any class:

Thanks for your suggestions. I have modified the code to create the `constrxpr` function and return literal.  For another suggestion on using the `constrxpr` variable, somehow, I don't see these getting optimized in different translation units by gcc compiler ( small pingy code to demonstrate that: https://www.onlinegdb.com/IjRRjLjZo ). But anyway preferred the `constexpr` function as we don't need to depend on linker/compiler optimization here.

Reply:
To summarise the available options:
 **option1**:  `constexpr static class methods`: The advantage of using these is that only those methods are generated (in .text segment) at compile time which are used across all the translation units. The rest of the unused methods are ignored.
**option2**: `constexpr static global functions in the namespace`: This will create a copy of the used functions in each translation unit while ignoring the rest of the unused functions. 
**option3**: `constexpt static variables in the namespace`: This will create a local copy of all the variables in each of the translation units including the header, and it seems that the gcc compiler doesn't optimize it.

The right solution comes from the C++17 `inline` variables, but `option1` seems to be the correct approach for C++11 which is now implemented in this PR. We can discuss if there are any issues with the approach.